Implement Air Hockey Envs in Fancy Gym

Thifhi opened the pull request:
This PR also introduces a new optional dependency "mushroom-rl" required for the air hockey environments.

Thifhi commented on fancy_gym/envs/mujoco/__init__.py:
```diff
@@ -9,3 +9,9 @@
+try:
+    from .air_hockey.air_hockey_env_wrapper import AirHockeyEnv
+except:
+    print("[FANCY GYM] Air Hockey not available (depends on mushroom-rl, dmc, mujoco)")
```
Addressed in 9aa572271faad6ebb587f210d7ad6f3b61cd50c0.

Thifhi commented on fancy_gym/envs/mujoco/air_hockey/air_hockey_env_wrapper.py:
```diff
@@ -0,0 +1,173 @@
+
+        if env_mode == None:
+            print("Please specify one of the environments in [3dof-hit, 3dof-defend, 7dof-hit, 7dof-defend, tournament]")
+            exit()
```
Addressed in 079225cd1bde4386c8851899590ff3695618977c.

Thifhi commented on fancy_gym/envs/__init__.py:
```diff
@@ -289,3 +289,9 @@
+    id='fancy/AirHockey-v0',
+    entry_point='fancy_gym.envs.mujoco:AirHockeyEnv',
+    max_episode_steps=45000
+)
```
The first part is addressed in 7e537447e1765d102af55695025e3eee18c78354.
For the second part I passed the empty list for mp_types in f102e495818f2ee6304d05485ab557de420e34cd. It is possible that a more experienced person who worked on air hockey with movement primitives might add support but for now I suggest that we leave it out.

Thifhi commented:
I also added a license and referred to the original project. If there are any issues about the way I put the license please let me know!

D-o-d-o-x reviewed (changes requested):
Our test suite has some failures regarding the new envs:

```
FAILED test/test_fancy_envs.py::test_step_fancy_functionality[fancy/AirHockey-tournament-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_functionality[fancy/AirHockey-7dof-hit-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_functionality[fancy/AirHockey-7dof-defend-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_functionality[fancy/AirHockey-3dof-hit-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_functionality[fancy/AirHockey-3dof-defend-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_determinism[fancy/AirHockey-tournament-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_determinism[fancy/AirHockey-7dof-hit-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_determinism[fancy/AirHockey-7dof-defend-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_determinism[fancy/AirHockey-3dof-hit-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
FAILED test/test_fancy_envs.py::test_step_fancy_determinism[fancy/AirHockey-3dof-defend-v0] - AttributeError: 'AirHockeyEnv' object has no attribute 'close'
```

The new envs do not implement .close(), which should e.g. close the Mujoco view when rendering was active or free other resources.

Also, per the new [gym spec](https://gymnasium.farama.org/api/env/#gymnasium.Env.render), the render() function in rendering_mode='human' should only be required to be called once initially, and subsequent calls to .step() should automatically update the rendering.
This isn't happening with the AirHockey envs; they require render() to be called after every step(), aligning with the old gym spec instead.

Lastly, regarding the max_episode_steps setting for the AirHockey envs. Is the 45000 step limit appropriate for versions other than 'tournament'? It's currently consuming a significant portion of our test suite's run time.

Thifhi commented:
1) I have now added an empty close method. As far as I know air hockey environments don't acquire any resources that need to be released.

2) I have adjusted render to behave per specs.

3) The step numbers were incorrect, thanks for spotting it. The regular envs are now 500 steps max and I have also reduced tournament step size to 15000.

D-o-d-o-x commented:
A tournaments can last 15min (afaik) and we render at 60fps (according to the envs metadata)?
Should that not need 15×60×60 = 54000 frames?
AirHockeyTournament.horizon is set to 45000. Is the fps number in the metadata wrong? (Would work out for 50fps)
The envs max_episode_steps is set to 15000.
Or do you configure a frame_skip somewhere, that I can't find?

.close no longer raises no exceptions, but also does not close open Mujoco-Viewers. The current implementation also leads to a small memory leak, as instances of mujoco are newer broken down. Can you somehow forward the .close to Mujoco? You might have a look [here](https://github.com/Farama-Foundation/Gymnasium/blob/main/gymnasium/envs/mujoco/mujoco_env.py) for that.

All test are now green except for test/test_fancy_envs.py:test_step_fancy_determinism[fancy/AirHockey-tournament-v0].
The tournament env does no behave deterministically under the same seed initialization. Can that be fixed?

D-o-d-o-x reviewed (approved):
Looks good. All tests are passing successfully. Additionally, I've checked for merge conflicts with the other open PRs and it seems clear. This means we have the flexibility to merge in any order. (I already merged the PR into the dev branch).

Thank you, @Thifhi, for addressing all the issues so quickly.
